avaframe/in3Utils/geoTrans.py

```diff
@@ -181,87 +181,83 @@ def resizeData(raster, rasterRef):
         return raster['rasterData'], rasterRef['rasterData']
 
 
-def getMeshXY(rasterDict, cellSizeNew=None):
-    """ Get x and y vectors for mesh with given number of rows and columns, lowerleftcenter and cellSize
+def remeshData(rasterDict, cellSizeNew, remeshOption='griddata', interpMethod='cubic', larger=True):
+    """ compute raster data on a new mesh with cellSize using the specified remeshOption.
 
-        Parameters
-        -----------
-        rasterDict: dict
-            'rasterData' : 2D numpy array of data
-            'header': class with info on ncols, nrows, csz, xllcenter, yllcenter, noDataValue
-
-        Returns
-        --------
-        x, y: 1D numpy arrays
-            vector of x and y values for mesh center coordinates
-        xExtent, yExtent: float
-            extent of mesh from first cell center to last cell center in x/y
-
-    """
-
-    header = rasterDict['header']
-    xllc = header['xllcenter']
-    yllc = header['yllcenter']
-    ncols = header['ncols']
-    nrows = header['nrows']
-    xExtent = (ncols-1) * header['cellsize']
-    yExtent = (nrows-1) * header['cellsize']
-    x = np.linspace(0, xExtent, ncols) + xllc
-    y = np.linspace(0, yExtent, nrows) + yllc
-
-    if cellSizeNew != None:
-
-        nColsNew = int(xExtent/cellSizeNew+1)
-        nRowsNew = int(yExtent/cellSizeNew+1)
-        xNew = np.linspace(0, (nColsNew-1)*cellSizeNew, nColsNew) + xllc
-        yNew = np.linspace(0, (nRowsNew-1)*cellSizeNew, nRowsNew) + yllc
-        diffExtentX = xExtent - (nColsNew-1)*cellSizeNew
-        diffExtentY = yExtent - (nRowsNew-1)*cellSizeNew
-        return x, y, xNew, yNew, diffExtentX, diffExtentY
-
-    else:
-        return x, y, xExtent, yExtent
-
-
-def remeshData(rasterFile, cellSize):
-    """ compute raster data on a new mesh with cellSize using scipy RectBivariateSpline
-
-        the new mesh is as big or smaller as the original mesh
+        remeshOption are to choose between 'griddata', 'interp2d' or 'RectBivariateSpline'
+        Only the 'griddata' works properly if the input data contains noData points,
+        'interp2d' or 'RectBivariateSpline' are faster but fail if input data contains noData points.
```

Comment:
does this throw an error?

Reply:
No it does not but the default option is 'griddata' and it is the one used in com1DFA. Should it check for nans if it is used with the other methods?

Reply:
If the methods do not work properly in this case and hence should not be used, I think it is a good idea to first check for nans and if they are there either use griddata and throw a warning or throw directly an error and say that the chosen method doesn't work as there are nans if this is the case